packages/health/lib/src/health_factory.dart

```diff
@@ -111,9 +111,12 @@ class HealthFactory {
     if (_platformType == PlatformType.ANDROID) _handleBMI(mTypes, mPermissions);
 
     List<String> keys = mTypes.map((e) => _enumToString(e)).toList();
-    final bool? isAuthorized = await _channel.invokeMethod(
+    final dynamic? isAuthorized = await _channel.invokeMethod(
         'requestAuthorization', {'types': keys, "permissions": mPermissions});
-    return isAuthorized ?? false;
```

Comment:
@chuong-pivot Chỗ này có nghĩa là cái package gốc nó handle sai dạng data trả về của hàm **requestAuthorization** đúng ko em. Ko phải boolean mà là dạng object gì đó mới đúng đúng ko ? 🤔

Reply:
e ko nhớ nữa

Reply:
@chuong-pivot Vậy ráng nhớ ra rồi giải thích đi em :D

Reply:
à cái này phòng trường hợp bên native xử lý lỗi, channel gửi ko đúng method,... nên mới bắt null ngay đó, xài channel thì chắc chắn p bắt null rồi a :D

Reply:
@chuong-pivot Vậy tức là dùng bool? thì ko bắt null được, mà phải dynamic? mới bắt được hả em ? 
Như vậy thì anh thấy em gần như đâu thay đổi gì logic của plugin gốc đâu, vậy sao lại giải quyết được vấn đề của em hay vậy :)

Reply:
hình như a hiểu sai ý e :v là xài cái channel để kêu bên native nhưng mà đâu thể chắc chắn 100% là nó kêu đúng func bên native hay là native nó sẽ thực thi rồi trả về mà bên flutter bắt được đâu anh.
![image](https://user-images.githubusercontent.com/110024125/181177159-2015d470-0454-4999-bcbe-647e01f25d7e.png)

Reply:
còn thay đổi logic plug-in thì anh qua bên plug-in a xem chứ bên này gọi thôi chứ có làm gì đâu, em thay đổi hàm requestAuthorization với thêm revokePermissions nha chứ lúc đầu nó ko có revokePermissions luôn ấy :)

Reply:
> còn thay đổi logic plug-in thì anh qua bên plug-in a xem chứ bên này gọi thôi chứ có làm gì đâu, em thay đổi hàm requestAuthorization với thêm revokePermissions nha chứ lúc đầu nó ko có revokePermissions luôn ấy :)

Ờ ngon em, sorry nãy anh nhìn sót cái diff của bên native (large diff nên nó default ko show ra)